nca/CoreDS/ConnectionSet.py

```diff
@@ -550,3 +570,109 @@ def get_non_tcp_connections():
+    @staticmethod
+    def conn_props_to_fw_rules(conn_props, cluster_info, peer_container, ip_blocks_mask,
+                               connectivity_restriction):
+        """
+        Build FWRules from the given ConnectivityProperties
+        :param ConnectivityProperties conn_props: properties describing allowed connections
+        :param ClusterInfo cluster_info: the cluster info
+        :param PeerContainer peer_container: the peer container
+        :param IpBlock ip_blocks_mask: IpBlock containing all allowed ip values,
+         whereas all other values should be filtered out in the output
+        :param Union[str,None] connectivity_restriction: specify if connectivity is restricted to
+               TCP / non-TCP , or not
+        :return: FWRules map
+        """
+        ignore_protocols = ProtocolSet()
+        if connectivity_restriction:
+            if connectivity_restriction == 'TCP':
+                ignore_protocols.add_protocol('TCP')
+            else:  # connectivity_restriction == 'non-TCP'
+                ignore_protocols = ProtocolSet.get_non_tcp_protocols()
+
+        fw_rules_map = defaultdict(list)
+        for cube in conn_props:
+            conn_cube = conn_props.get_connectivity_cube(cube)
+            src_peers = conn_cube["src_peers"]
+            if not src_peers:
+                src_peers = peer_container.get_all_peers_group(True)
+            conn_cube.unset_dim("src_peers")
+            dst_peers = conn_cube["dst_peers"]
+            if not dst_peers:
+                dst_peers = peer_container.get_all_peers_group(True)
+            conn_cube.unset_dim("dst_peers")
+            if IpBlock.get_all_ips_block() != ip_blocks_mask:
+                src_peers.filter_ipv6_blocks(ip_blocks_mask)
+                dst_peers.filter_ipv6_blocks(ip_blocks_mask)
+            protocols = conn_cube["protocols"]
+            conn_cube.unset_dim("protocols")
+            if not conn_cube.has_active_dim() and (protocols.is_whole_range() or protocols == ignore_protocols):
+                conns = ConnectionSet(True)
+            else:
+                conns = ConnectionSet()
+                protocol_names = ProtocolSet.get_protocol_names_from_interval_set(protocols) if protocols else ['TCP']
+                for protocol in protocol_names:
+                    if conn_cube.has_active_dim():
+                        conns.add_connections(protocol, ConnectivityProperties.make_conn_props(conn_cube))
+                    else:
+                        if ConnectionSet.protocol_supports_ports(protocol):
+                            conns.add_connections(protocol, ConnectivityProperties.make_all_props())
+                        elif ConnectionSet.protocol_is_icmp(protocol):
+                            conns.add_connections(protocol, ConnectivityProperties.make_all_props())
+                        else:
+                            conns.add_connections(protocol, True)
```

Comment:
the first two branches can be merged to one

Reply:
Done